components/merino_client/Cargo.toml

```diff
@@ -0,0 +1,26 @@
+[package]
+name = "merino_client"
+edition = "2021"
+version = "0.1.0"
+authors = ["contextual-services-team@mozilla.com"]
+license = "MPL-2.0"
+exclude = ["/android", "/ios"]
+
+[dependencies]
+anyhow = "1.0"
+error-support = { path = "../support/error" }
+log = "0.4"
+serde = "1"
+serde_derive = "1"
+serde_json = "1"
+thiserror = "1.0"
+uniffi = "^0.21"
+uniffi_macros = "^0.21"
+url = "2.1" # mozilla-central can't yet take 2.2 (see bug 1734538)
+viaduct = { path = "../viaduct" }
```

Comment:
Viaduct is the library that we use to make HTTP requests.

* On Android, it uses "Necko", the Gecko networking stack—Android Components sets up the Viaduct backend at launch, then Viaduct calls into Android Components, a-c makes the request via GeckoView, and passes the response back through the different layers to Viaduct.
* On iOS (and in tests), it uses the same system network stack as WebKit, via `viaduct-reqwest`.
* On Desktop, it makes requests using Necko directly.